File: C/Lista 1/06.c
/* Escreva um algoritmo que lê um valor n inteiro e positivo e que calcula a seguinte soma:
S = 1 + 1/2 + 1/3 + 1/4 + ... + 1/n
O algoritmo deve escrever cada termo gerado e o valor final de S. */

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(){
 
    int n;
    float soma = 0;
    float aux = 0;
    printf("\nDigite um valor inteiro: ");
    scanf(" %i", &n);

    while(n > 0){
       soma += 1/(aux++);
       n--;
    }    
    printf("\nO valor da soma e: %.2f", soma);
    
    return 0;
}

File: C#/Lista 1/06.cs
/* Escreva um algoritmo que lê um valor n inteiro e positivo e que calcula a seguinte soma:
S = 1 + 1/2 + 1/3 + 1/4 + ... + 1/n
O algoritmo deve escrever cada termo gerado e o valor final de S. */

using System;

class Program {
    public static void Main(string[] args){
        int n = 0;
        float soma = 0;
        float aux = 0;

        Console.WriteLine("Digite um valor inteiro: ");
        n = int.Parse(Console.ReadLine());

        while(n >= 0){
            float aux2 = 0;
            aux2 = aux++;
            soma += (float)1/aux2;
            n--;
        }

        Console.WriteLine("O valor da soma é: " + soma);
    }
}